src/test/compile-fail/E0050.rs

```diff
@@ -16,6 +16,7 @@ struct Bar;
 
 impl Foo for Bar {
     fn foo(&self) -> bool { true } //~ ERROR E0050
+                                   //~| NOTE expected 2 parameters
```

Comment:
Since you're changing the spans, can you move this to being a UI test (src/test/ui) instead?